media-sound/sinus/patches/sinus-0.1.3.patchset

```diff
@@ -0,0 +1,60 @@
+From 0af08e2e3c3dcc85b2be23e09e71c78aec95dac5 Mon Sep 17 00:00:00 2001
+From: Bach Nguyen <bach5000@gmail.com>
+Date: Thu, 22 Nov 2018 11:51:15 +0000
+Subject: Fix
+
+
+diff --git a/sinus.c b/sinus.c
+index aaab435..7012ad4 100644
+--- a/sinus.c
++++ b/sinus.c
+@@ -235,31 +235,34 @@ void wave_to_little_endian(struct wave_t *wave)
+ #ifdef __LITTLEENDIAN__
+   return;
+ #else
+-  int i;
+-  void *p;
+ 
++  int i = 0;
+   switch(wave->point_size)
+   {
+     case 1:
+       return;
+     case 2:
+-      p = wave->data;
+-      for(i=0; i<wave->data_size; i+=2)
++    {
++      INT16 *p;
++      for(p = wave->data; i < wave->data_size; i += 2)
+       {
+-        dprintf((stderr, "host -> le: %04X ->", (UINT16)*((UINT16 *)(p+i))));
+-        (INT16)*((UINT16 *)(p+i)) = HOST_TO_LENDIAN_INT16(*((UINT16 *)(p+i)));
+-        dprint((stderr, " %04X\n", (UINT16)*((UINT16 *)(p+i))));
++        dprintf((stderr, "host -> le: %04X ->", *p));
++        *p = HOST_TO_LENDIAN_INT16(*p);
++        dprint((stderr, " %04X\n", *p));
+       }
+       return;
++    }
+     case 4:
+-      p = wave->data;
+-      for(i=0; i<wave->data_size; i+=4)
++    {
++      INT32 *p;
++      for(p = wave->data; i < wave->data_size; i += 2)
```

Comment:
It should be `i += 4` here, but let’s use `i += wave->point_size` to make the code more robust.